src/test/java/org/sagebionetworks/bridge/exporter/integration/WeeklyAdherenceReportWorkerProcessorTest.java

```diff
@@ -0,0 +1,149 @@
+        userId = user.getUserId();
+        
+        admin = TestUserHelper.getSignedInAdmin();
+        scheduleApi = admin.getClient(SchedulesV2Api.class);
+        adherenceApi = admin.getClient(StudyAdherenceApi.class);
+        assessmentApi = admin.getClient(AssessmentsApi.class);
+        participantApi = admin.getClient(ParticipantsApi.class);
+        
+        // We need to create a schedule in study1 for this user if it doesn't exist.
+        try {
+            scheduleApi.getScheduleForStudy(STUDY_ID_1).execute().body();
+        } catch(EntityNotFoundException e) {
+            String assessmentId = RandomStringUtils.randomAlphabetic(10);
+            
+            assessment = new Assessment()
+                    .title(getClass().getSimpleName() + " " + assessmentId)
+                    .osName("Universal")
+                    .ownerId(SAGE_ID)
+                    .identifier(assessmentId);
+            assessment = assessmentApi.createAssessment(assessment).execute().body();
+            
+            AssessmentReference2 ref = new AssessmentReference2()
+                    .guid(assessment.getGuid())
+                    .appId(TEST_APP_ID)
+                    .revision(5)
+                    .title("A title")
+                    .identifier(assessment.getIdentifier());
+            
+            Session session = new Session();
+            session.setName("Simple session");
+            session.addStartEventIdsItem("created_on");
+            session.setPerformanceOrder(SEQUENTIAL);
+            session.addAssessmentsItem(ref);
+            session.addTimeWindowsItem(new TimeWindow().startTime("08:00").expiration("P3D"));
+
+            schedule = new Schedule2();
+            schedule.setName("Test Schedule [WeeklyAdherenceReportWorkerProcessorTest]");
+            schedule.setDuration("P1W");
+            schedule.addSessionsItem(session);
+
+            schedule = scheduleApi.saveScheduleForStudy(STUDY_ID_1, schedule).execute().body();
+        }
+    }
+    
+    @AfterMethod
+    public void afterMethod() throws IOException {
+        if (schedule != null) {
+            scheduleApi.deleteSchedule(schedule.getGuid()).execute().body();
+            assessmentApi.deleteAssessment(assessment.getGuid(), true).execute().body();
+        }
+        if (user != null) {
+            user.signOutAndDeleteUser();
+        }
+    }
+
+    @Test
+    public void requestCaching() throws Exception {
+        String requestText = "{\"service\":\"WeeklyAdherenceReportWorker\", \"body\":{\"appId\":\"api\",\"studyId\":\"study1\"}}";
+        ObjectNode requestNode = (ObjectNode) DefaultObjectMapper.INSTANCE.readTree(requestText);
+
+        sqsHelper.sendMessageAsJson(workerSqsUrl, requestNode, 0);
+        
+        // Wait. Let the worker do its thing.
+        Thread.sleep(8000L);
+        
+        // This should return our user...
+        assertTrue( findUser() );
```

Comment:
Should there be a report generated that we should be looking at?

Reply:
This was a dumb one on my part, this helper method *did* look for the existence of the report, and I changed it. I've reverted it, and renamed the method to what it actually does (it looks for the report, not the user).